composer.json

```diff
@@ -40,6 +40,7 @@
     "require-dev": {
         "sensio/generator-bundle": "^3.0",
         "symfony/phpunit-bridge": "^3.0",
+        "symfony/web-server-bundle": "^3.3",
```

Comment:
This should also have made some changes in the composer.lock file and should be included in the commit.

Reply:
I think it did, but it looks a bit strange: it added the line "symfony/web-server-bundle": "self.version", into replace() in the symfony/symfony block

Reply:
I'm not sure I follow you. What do you mean by "the symfony/symfony block" ?

Reply:
Sorry. I mean the only difference in composer.lock is that single line "symfony/web-server-bundle": "self.version". The line is in the code block for main symfony dependencies, but it's not under reqiure{} (it's under replace {})

I have deleted the files (json and lock), re-installed & updated composer.. it's still the same.

Can I manually add this under require{} like:

{
            "name": "symfony/symfony",
            "version": "v3.3.9",
            "source": {
                "type": "git",
                "url": "https://github.com/symfony/symfony.git",
                "reference": "a9d2f68ae9946000e2bcc3403d218ec0124f992a"
            },
            "dist": {
                "type": "zip",
                "url": "https://api.github.com/repos/symfony/symfony/zipball/a9d2f68ae9946000e2bcc3403d218ec0124f992a",
                "reference": "a9d2f68ae9946000e2bcc3403d218ec0124f992a",
                "shasum": ""
            },
            "require": {
                "doctrine/common": "~2.4",
                "ext-xml": "*",
                "fig/link-util": "^1.0",
                "php": "^5.5.9|>=7.0.8",
                "psr/cache": "~1.0",
                "psr/container": "^1.0",
                "psr/link": "^1.0",
                "psr/log": "~1.0",
                "psr/simple-cache": "^1.0",
                "symfony/polyfill-apcu": "~1.1",
                "symfony/polyfill-intl-icu": "~1.0",
                "symfony/polyfill-mbstring": "~1.0",
                "symfony/polyfill-php56": "~1.0",
                "symfony/polyfill-php70": "~1.0",
                "symfony/polyfill-util": "~1.0",
                "twig/twig": "~1.34|~2.4"
                "symfony/web-server-bundle": "~3.0",
            },

Reply:
So, if I've got this right, it looks like you did a `composer require --dev symfony/web-server-bundle` or something like it. That's the right thing to do here. Composer (not you!) made some changes to the composer.lock file, which is also the right thing.

Composer manages the .lock file - we never need to edit it. It has all the versioning information for the dependencies and the their dependencies and so on.  It's a serialized data structure that contains enough information to recreate all the dependencies exactly. 

Even though we never change it, we do keep it in version control so that we can recreate the environment as exactly as possible if we need to.

That's a long way of saying, don't change the composer.lock file by hand, let composer do it. Just commit any changes that get made to the lock file.

Reply:
got it! thanks!